### PR TITLE
#2017 - Upgrade npm packages to remove warnings accross all projects

### PR DESCRIPTION
**Pull request by andrepestana-aot**
Upgraded npm packages to fix the warnings below when running `npm ci`:
image

### Soap issues
Updated soap version to 1.0.0 which fixed:
  - npm WARN deprecated request@2.88.2: request has been deprecated
  - npm WARN deprecated har-validator@5.1.5: this library is no longer supported

### Zeebe Node issues
@types/keyv@4.2.0 is a dependency of got@11.8.5 that is a dependency of zeebe-node@8.1.4. 
image

The zeebee-node version won't be updated for now because of some compatibility issues. However, even updating to zeebe-node@8.2.4 as a test did not update `got` version and it continues displaying the warning for `@types/keyv`. Updating `zeebe-node` to 8.2.4 updates `uuid` and makes the warning to disappear though.

Also, we can notice that `fast-xml-parser` is a dependency of `zeebe-node`. There are 2 dependabot alerts for this package's version:
https://github.com/bcgov/SIMS/security/dependabot/117
https://github.com/bcgov/SIMS/security/dependabot/116 
Updating `zeebe-node` to version 8.2.4 will fix them as `fast-xml-parser` version will be 4.2.4.

### Unnecessary packages
Removed the packages below as they are not necessary any more:
  - @types/bull@4.10.0
  - @types/jwt-decode@3.1.0

### Super Test
`supertest` was updated to "^6.3.3" and removed the warning for `superagent`  and  `formidable` as they are its dependencies.

### Class Transformer
Updated to version 0.5.1.

### Conclusion
There are still 2 warnings for `@types/keyv@4.2.0` and `uuid@3.4.0`. The `uuid@3.4.0` one will be fixed when  `zeebe-node` is updated to version 8.2.4 and the `@types/keyv@4.2.0` one should be fixed in some `zeebe-node` future version.
image

## Info
To check the dependency tree, run:
`npm ls -a > dependencies.txt`

### CODE REVIEW

File: sources/packages/backend/package.json [hunk 1]
```diff
@@ -85,13 +85,11 @@
     "@nestjs/cli": "^9.3.0",
     "@nestjs/schematics": "^9.0.3",
     "@nestjs/testing": "^9.1.2",
-    "@types/bull": "^4.10.0",
```
Comment (ann-aot) [inline sources/packages/backend/package.json:88]: just for info, was this and line 94 deleted as part of any package update

**Review by ann-aot — APPROVED** [review]
LGTM 👍 Added a minor question

File: sources/packages/backend/package.json (hunk 1 above)
Comment (andrepestana-aot) [reply sources/packages/backend/package.json:88]: As the ticket description, there was a warning for that. This package `@types/bull` is not needed anymore as the own warning mentioned.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍 
Thanks for adding the great description. It can be useful if we need to justify the current security issues reported by GitHub right now.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the all fixes. Great job.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrepestana-aot

**Pull request merged**